### PR TITLE
chore(router): switch to vitest

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
This PR changes the router package to use vitest over jest for the unit tests.

### CODE REVIEW

File: packages/router/src/__tests__/useMatch.test.tsx [hunk 1]
```diff
@@ -55,7 +54,7 @@ describe('useMatch', () => {
       </LocationProvider>,
     )
 
-    expect(getByText(/Dunder Mifflin/)).toHaveStyle('color: green')
+    expect(getByText(/Dunder Mifflin/)).toHaveStyle('color: rgb(0, 128, 0)')
```
Comment (Tobbe) [inline packages/router/src/__tests__/useMatch.test.tsx:57]: Huh?! This was an unexpected change from going from jest to vitest

Comment (Josh-Walker-GM) [reply packages/router/src/__tests__/useMatch.test.tsx:57]: Yeah. Not sure why it's different. I did a quick search but nothing immediately came up.

File: packages/router/vitest.setup.mts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import '@testing-library/jest-dom/vitest'
+
+import { afterEach, vi } from 'vitest'
+import { cleanup } from '@testing-library/react'
+
+// @ts-expect-error
+globalThis.scrollTo = vi.fn()
```
Comment (Josh-Walker-GM) [inline packages/router/vitest.setup.mts:7]: @Tobbe I suspect you'll not like this. I haven't tried to type a mock to scrollTo before and whatever I try fails to make TS happy.

File: packages/router/src/__tests__/useMatch.test.tsx (hunk 1 above)
Comment (Tobbe) [reply packages/router/src/__tests__/useMatch.test.tsx:57]: Should we at least make the input match?

File: packages/router/vitest.setup.mts (hunk 2 above)
Comment (Tobbe) [reply packages/router/vitest.setup.mts]: Fixed in [91e85c1](https://github.com/redwoodjs/redwood/pull/10674/commits/91e85c1e84e9fd566d549677c196cd636f522f2f)

File: packages/router/src/__tests__/useMatch.test.tsx (hunk 1 above)
Comment (Tobbe) [reply packages/router/src/__tests__/useMatch.test.tsx]: I switched to using hex codes in [0575e22](https://github.com/redwoodjs/redwood/pull/10674/commits/0575e22c3f1f7df91b740f5cb5fd0be08000cdac)

Happy to revert if you don't like that. Or change in other ways

**Pull request merged**